src/main/java/com/microsoft/graph/content/BatchResponseContent.java

```diff
@@ -0,0 +1,228 @@
+        }
+        return responseHandler.handleResponseAsync(response, apiErrorMappings);
+    }
+    /**
+     * Gets the response within the batch response via specified id.
+     * @param requestId The id of the request.
+     * @param factory The factory to use when deserializing the response body.
+     * @return The response within the batch response via specified id, null if not found.
+     * @param <T> The type of the response body.
+     */
+    @Nullable
+    public <T extends Parsable> CompletableFuture<T> getResponseByIdAsync(@Nonnull String requestId, @Nonnull ParsableFactory<T> factory) {
+        return this.getResponseByIdAsync(requestId, new ResponseBodyHandler<>(factory));
+    }
+    /**
+     * Gets the response within the batch response via specified id.
+     * @param requestId The id of the request.
+     * @return The response within the batch response via specified id as an InputStream, null if not found.
+     */
+    @Nullable
+    @SuppressFBWarnings
+    public CompletableFuture<InputStream> getResponseStreamByIdAsync(@Nonnull String requestId) {
+        Response response = getResponseByIdAsync(requestId).join();
+        if(response != null && response.body() != null) {
+            try{
+                InputStream in = response.body().byteStream();
+                ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(ByteStreams.toByteArray(in));
+                return CompletableFuture.completedFuture(byteArrayInputStream);
+            } catch (IOException e) {
+                CompletableFuture<InputStream> exception = new CompletableFuture<>();
+                exception.completeExceptionally(e);
+                return exception;
+            }
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+    /**
+     * Get the next link of the batch response.
+     * @return The next link of the batch response.
+     */
+    @Nullable
+    public CompletableFuture<String> getNextLinkAsync() {
+        jsonBatchResponseObject = jsonBatchResponseObject != null ? jsonBatchResponseObject : getBatchResponseContentAsync().join();
+        if(jsonBatchResponseObject != null) {
+            JsonElement nextLinkElement = jsonBatchResponseObject.get(CoreConstants.Serialization.ODATA_NEXT_LINK);
+            if (nextLinkElement != null && nextLinkElement.isJsonPrimitive()) {
+                return CompletableFuture.completedFuture(nextLinkElement.getAsString());
+            }
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+    @SuppressFBWarnings
+    private CompletableFuture<JsonObject> getBatchResponseContentAsync() {
+        if (this.batchResponse.body() != null && this.batchResponse.body().contentType() != null) {
+            InputStream in = this.batchResponse.body().byteStream();
+            return CompletableFuture.completedFuture(JsonParser.parseReader(new InputStreamReader(in, StandardCharsets.UTF_8)).getAsJsonObject());
```

Comment:
input stream reader is closeable, use the try syntax

Reply:
done